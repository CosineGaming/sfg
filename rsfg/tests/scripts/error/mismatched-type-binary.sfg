fn main()
	if 5 == 5.0
		panic()
	var bool_times = false * true
	var float_times = 2.0 * 3.0
	var int_and = 5 && 6
	var float_and = 5f && 6f
	var str_eq = "this" == "this" // TODO support this instead of error
	var str_minus = "one" - "two"

