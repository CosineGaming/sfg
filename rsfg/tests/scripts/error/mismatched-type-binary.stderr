[ERROR] mismatched type, expected int at 2:5, got float at 2:10
[ERROR] no operation Times for bool at 4:25
[ERROR] no operation Times for float at 5:24
[ERROR] no operation And for int at 6:18
[ERROR] no operation And for float at 7:21
[ERROR] no operation Equal for str at 8:22
[ERROR] no operation Minus for str at 9:24